backend/danswer/background/celery/celery_app.py

```diff
@@ -411,26 +456,37 @@ def check_for_vespa_sync_task() -> None:
 
 
 @celery_app.task(
-    name="check_for_cc_pair_deletion_task",
+    name="check_for_connector_deletion_task",
     soft_time_limit=JOB_TIMEOUT,
 )
-def check_for_cc_pair_deletion_task() -> None:
-    """Runs periodically to check if any deletion tasks should be run"""
-    with Session(get_sqlalchemy_engine()) as db_session:
-        # check if any cc pairs are up for deletion
-        cc_pairs = get_connector_credential_pairs(db_session)
-        for cc_pair in cc_pairs:
-            if should_kick_off_deletion_of_cc_pair(cc_pair, db_session):
-                task_logger.info(
-                    f"Deleting the {cc_pair.name} connector credential pair"
-                )
+def check_for_connector_deletion_task() -> None:
+    r = redis_pool.get_client()
 
-                cleanup_connector_credential_pair_task.apply_async(
-                    kwargs=dict(
-                        connector_id=cc_pair.connector.id,
-                        credential_id=cc_pair.credential.id,
-                    ),
+    lock_beat = r.lock(
+        DanswerRedisLocks.CHECK_CONNECTOR_DELETION_BEAT_LOCK,
+        timeout=CELERY_VESPA_SYNC_BEAT_LOCK_TIMEOUT,
```

Comment:
is 60 seconds enough to create a job for a very large connector (e.g. 100k docs)?

Reply:
60 second is just the initial timeout ... the lock is actually passed down and reacquired frequently while iterating through rows to generate tasks, so this should be fine.